### PR TITLE
Add heroku archetype project

### PR DESCRIPTION
**Pull request by aklish**
## Description
First of several PRs to improve the documentation and provide an archetype setup for an elide project.

## Motivation and Context
Elide getting started docs are pretty dated - and not as helpful as they ought to be.

## How Has This Been Tested?
Tested with Heroku deployment.

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: app.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "Elide Sample Service",
+  "description": "An example Elide service with an in-memory DB",
+  "repository": "https://github.com/yahoo/elide/tree/master/elide-example/elide-blog-example",
+  "logo": "https://elide.io//assets/images/elide-white-logo.png",
+  "keywords": ["java", "elide", "graphql", "json-api"]
```
Comment (jkusa) [inline app.json]: Can we add 
```
"addons": [ "heroku-postgresql" ]
```

**Comment by aklish** [comment]
Do you think we should do Postgres instead of H2?

**Comment by jkusa** [comment]
@aklish 

> If an app has a free web dyno, and that dyno receives no web traffic in a 30-minute period, it will sleep.

If we configure Postgres, user's data will persist across sessions. Also, they will be able to inspect the db data via the heroku-postgresql tools.

File: elide-example/elide-blog-example/pom.xml [hunk 2]
```diff
@@ -1,148 +1,72 @@
...
-        <groupId>com.yahoo.elide</groupId>
-        <artifactId>elide-example-parent-pom</artifactId>
-        <version>4.5.3-SNAPSHOT</version>
-    </parent>
+    <groupId>com.yahoo.elide</groupId>
+    <version>4.5.3-SNAPSHOT</version>
 
     <properties>
-        <mysql.data.directory>${project.build.directory}/mysql-data</mysql.data.directory>
+        <jar.finalName>testapp</jar.finalName>
```
Comment (jkusa) [inline elide-example/elide-blog-example/pom.xml]: what about `<jar.finalName>elide-blog-example</jar.finalName>`?

**Comment by aklish** [comment]
Going to add H2 IT tests (with the changes to use Postgres).  Then will make reviewable again.

File: README.md [hunk 3]
```diff
@@ -75,7 +75,13 @@ More information about Elide can be found at [elide.io](http://elide.io/).
 
 ## Install
 
-The fastest way to get started with Elide is to use [elide-standalone](https://github.com/yahoo/elide/tree/master/elide-standalone) which allows you to quickly setup a local instance
+To try out an Elide example service (with an in-memory database), you can deploy via Heroku.
```
Comment (jkusa) [inline README.md]: `(with an in-memory database)` is this true?

File: elide-example/elide-blog-example/pom.xml (hunk 2 above)
Comment (jkusa) [reply elide-example/elide-blog-example/pom.xml]: ?

**Pull request merged**